packages/session-replay-browser/src/config/joined-config.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  SessionReplaySessionIDBStore as AmplitudeSessionReplaySessionIDBStore,
+  SessionReplayOptions,
+} from '../typings/session-replay';
+import { SessionReplayLocalConfig } from './local-config';
+import { SessionReplayRemoteConfigFetch } from './remote-config';
+import {
+  SessionReplayLocalConfig as ISessionReplayLocalConfig,
+  SessionReplayRemoteConfigFetch as ISessionReplayRemoteConfigFetch,
+  SessionReplayJoinedConfig,
+} from './types';
+
+export class SessionReplayJoinedConfigGenerator {
+  localConfig: ISessionReplayLocalConfig;
+  remoteConfigFetch: ISessionReplayRemoteConfigFetch;
+  sessionIDBStore: AmplitudeSessionReplaySessionIDBStore;
+
+  constructor(apiKey: string, options: SessionReplayOptions, sessionIDBStore: AmplitudeSessionReplaySessionIDBStore) {
+    this.sessionIDBStore = sessionIDBStore;
+    this.localConfig = new SessionReplayLocalConfig(apiKey, options);
+    this.remoteConfigFetch = new SessionReplayRemoteConfigFetch({
+      localConfig: this.localConfig,
+      sessionIDBStore: this.sessionIDBStore,
+    });
+  }
+
+  async generateJoinedConfig(sessionId: number) {
+    const config: SessionReplayJoinedConfig = { ...this.localConfig };
+    // Special case here as optOut is implemented via getter/setter
+    config.optOut = this.localConfig.optOut;
```

Comment:
[nit] might consider a static method here that takes local and remote config. i.e something like
`generateJoinedConfig(localConfig, remoteConfig) -> config`. more explicit and easier to test. right now `config` is assumed to be local config.

Reply:
True, what I was hoping though is that the consumer (session-replay.ts) would not have to know the difference between local and remote config - so it only has to deal with a single config that is generated by this method. The localConfig comes from the options passed to the SDK, and is created in this class, not exposed outside of this class